notebooks/config.ipynb

```diff
@@ -964,12 +971,831 @@
+      " |      This is invoked early on by abc.ABCMeta.__subclasscheck__().\n",
+      " |      It should return True, False or NotImplemented.  If it returns\n",
+      " |      NotImplemented, the normal algorithm is used.  Otherwise, it\n",
+      " |      overrides the normal algorithm (and the outcome is cached).\n",
+      " |\n",
+      " |  ----------------------------------------------------------------------\n",
+      " |  Class methods inherited from collections.abc.Iterable:\n",
+      " |\n",
+      " |  __class_getitem__ = GenericAlias(...)\n",
+      " |      Represent a PEP 585 generic type\n",
+      " |\n",
+      " |      E.g. for t = list[int], t.__origin__ is list and t.__args__ is (int,).\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.INIConfig)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "91a793e9-d1d1-41bb-84fd-c0c279606473",
+   "metadata": {},
+   "source": [
+    "An object can be initialized by providing a config either as a Python `dict` or a <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> to the file.\n",
+    "<!--cell 74-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 38,
+   "id": "0e8d6c8a-cc3c-49d7-9b97-32187cd5f754",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[fruit count]\n",
+      "apples = 3\n",
+      "grapes = {{ grape_count }}\n",
+      "kiwis = 2\n"
+     ]
+    }
+   ],
+   "source": [
+    "fruits = config.INIConfig(\n",
+    "    config=Path('fixtures/config/fruit-config.ini')\n",
+    ")\n",
+    "print(fruits)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1a50a362-d063-4ae1-9b6a-d98d0bf84c4f",
+   "metadata": {},
+   "source": [
+    "### Comparing Config Objects\n",
+    "\n",
+    "The `compare_config()` method compares two config `dict`s and returns `True` when they match and `False` otherwise. Two config `dict`s can be passed to the `dict1` and `dict2` parameters or, to compare a config `dict` to the class config, `dict2` can be left unspecified or set to `None`.\n",
+    "<!--cell 76-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 39,
+   "id": "73925576-6b07-4af6-a9b0-7bd6f4235987",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "True"
+      ]
+     },
+     "execution_count": 39,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "fruits.compare_config(\n",
+    "    dict1={'fruit count':{'apples':'3', 'grapes':'8', 'kiwis':'1'}},\n",
+    "    dict2={'fruit count':{'apples':'3', 'grapes':'8', 'kiwis':'1'}}\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3caa77b7-2eca-498b-841f-b37c91068974",
+   "metadata": {},
+   "source": [
+    "When there are differences between the two configs, `False` is returned. When a logger is initialized, the values that differ are displayed.\n",
```

Comment:
Here, we should mention that when a second dictionary is not provided, the comparison happens between `dict1` and the object itself -- `fruit` in this example.

Reply:
Good point. I had mentioned this in a previous cell, but I think it's better to have this information here.

Reply:
I'm sorry I missed that earlier on. Thanks for pulling it into this space for folks who don't retain info for very long, like me. :)